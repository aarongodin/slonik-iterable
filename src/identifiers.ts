import { sql, ListSqlTokenType } from "slonik"

export function fromArray(identifiers: string[], tableName?: string): ListSqlTokenType {
  return sql.join(
    identifiers.map((ident) => {
      const idents = [ident]
      if (tableName) {
        idents.unshift(tableName)
      }
      return sql.identifier(idents)
    }),
    sql`, `,
  )
}

export function fromObject(obj: Record<string, any>, tableName?: string): ListSqlTokenType {
  return fromArray(Object.keys(obj), tableName)
}
