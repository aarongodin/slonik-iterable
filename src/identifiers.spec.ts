import * as identifiers from "./identifiers"

describe("identifier helpers", () => {
  describe("fromArray()", () => {
    it("should return the list of identifiers without a table name", () => {
      const actual = identifiers.fromArray(["first", "secondIdent", "third_ident"])
      expect(actual).toMatchSnapshot()
    })

    it("should return the list of identifiers with a table name", () => {
      const actual = identifiers.fromArray(["first", "secondIdent", "third_ident"], "my_table")
      expect(actual).toMatchSnapshot()
    })
  })

  describe("fromObject()", () => {
    it("should return the list of identifiers based on object keys", () => {
      const actual = identifiers.fromObject({
        first: "first!",
        secondIdent: "second",
        third_ident: "thirdsies",
      })

      expect(actual).toMatchSnapshot()
    })

    it("should return the list of identifiers based on object keys including a table name", () => {
      const actual = identifiers.fromObject(
        {
          first: "first!",
          secondIdent: "second",
          third_ident: "thirdsies",
        },
        "my_table",
      )

      expect(actual).toMatchSnapshot()
    })
  })

  describe("fromSet()", () => {
    it("should return the list of identifiers without a table name", () => {
      const actual = identifiers.fromSet(new Set(["first", "secondIdent", "third_ident"]))
      expect(actual).toMatchSnapshot()
    })

    it("should return the list of identifiers with a table name", () => {
      const actual = identifiers.fromSet(new Set(["first", "secondIdent", "third_ident"]), "my_table")
      expect(actual).toMatchSnapshot()
    })
  })
})
