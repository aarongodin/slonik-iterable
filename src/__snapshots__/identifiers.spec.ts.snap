// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`identifier helpers fromArray() should return the list of identifiers with a table name 1`] = `
Object {
  "glue": Object {
    "sql": ", ",
    "type": "SLONIK_TOKEN_SQL",
    "values": Array [],
  },
  "members": Array [
    Object {
      "names": Array [
        "my_table",
        "first",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
    Object {
      "names": Array [
        "my_table",
        "secondIdent",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
    Object {
      "names": Array [
        "my_table",
        "third_ident",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
  ],
  "type": "SLONIK_TOKEN_LIST",
}
`;

exports[`identifier helpers fromArray() should return the list of identifiers without a table name 1`] = `
Object {
  "glue": Object {
    "sql": ", ",
    "type": "SLONIK_TOKEN_SQL",
    "values": Array [],
  },
  "members": Array [
    Object {
      "names": Array [
        "first",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
    Object {
      "names": Array [
        "secondIdent",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
    Object {
      "names": Array [
        "third_ident",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
  ],
  "type": "SLONIK_TOKEN_LIST",
}
`;

exports[`identifier helpers fromObject() should return the list of identifiers based on object keys 1`] = `
Object {
  "glue": Object {
    "sql": ", ",
    "type": "SLONIK_TOKEN_SQL",
    "values": Array [],
  },
  "members": Array [
    Object {
      "names": Array [
        "first",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
    Object {
      "names": Array [
        "secondIdent",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
    Object {
      "names": Array [
        "third_ident",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
  ],
  "type": "SLONIK_TOKEN_LIST",
}
`;

exports[`identifier helpers fromObject() should return the list of identifiers based on object keys including a table name 1`] = `
Object {
  "glue": Object {
    "sql": ", ",
    "type": "SLONIK_TOKEN_SQL",
    "values": Array [],
  },
  "members": Array [
    Object {
      "names": Array [
        "my_table",
        "first",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
    Object {
      "names": Array [
        "my_table",
        "secondIdent",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
    Object {
      "names": Array [
        "my_table",
        "third_ident",
      ],
      "type": "SLONIK_TOKEN_IDENTIFIER",
    },
  ],
  "type": "SLONIK_TOKEN_LIST",
}
`;
